React/CxxBridge/RCTCxxBridge.mm

```diff
@@ -1328,25 +1328,42 @@ - (void)executeApplicationScript:(NSData *)script
       [self->_performanceLogger markStopForTag:RCTPLRAMBundleLoad];
       [self->_performanceLogger setValue:scriptStr->size() forTag:RCTPLRAMStartupCodeSize];
       if (self->_reactInstance) {
-        auto registry = RAMBundleRegistry::multipleBundlesRegistry(std::move(ramBundle), JSIndexedRAMBundle::buildFactory());
-        self->_reactInstance->loadRAMBundle(std::move(registry), std::move(scriptStr),
+        self->_reactInstance->loadRAMBundle(std::move(ramBundle), std::move(scriptStr),
                                             sourceUrlStr.UTF8String, !async);
       }
     } else if (self->_reactInstance) {
       self->_reactInstance->loadScriptFromString(std::make_unique<NSDataBigString>(script),
                                                  sourceUrlStr.UTF8String, !async);
     } else {
-      std::string methodName = async ? "loadApplicationScript" : "loadApplicationScriptSync";
+      std::string methodName = async ? "loadBundle" : "loadBundleSync";
       throw std::logic_error("Attempt to call " + methodName + ": on uninitialized bridge");
     }
   }];
 }
 
 - (void)registerSegmentWithId:(NSUInteger)segmentId path:(NSString *)path
 {
-  if (_reactInstance) {
-    _reactInstance->registerBundle(static_cast<uint32_t>(segmentId), path.UTF8String);
-  }
+    __weak RCTCxxBridge *weakSelf = self;
+    NSURL *pathURL = [NSURL URLWithString:path];
+    dispatch_group_t group = dispatch_group_create();
+    
+    dispatch_group_enter(group);
+    [RCTJavaScriptLoader loadBundleAtURL:pathURL onProgress:^(RCTLoadingProgress *progressData) {} onComplete:^(NSError *error, RCTSource *source) {
+    if (error) {
+      [weakSelf handleError:error];
+      return;
+    }
+    
+    NSData *sourceCode = source.data;
+    __strong RCTCxxBridge *strongSelf = weakSelf;
+    if (strongSelf->_reactInstance) {
+      [strongSelf executeApplicationScript:sourceCode url:pathURL async:NO];
+    }
+    dispatch_group_leave(group);
+        
+    }];
+    
+    dispatch_group_wait(group, DISPATCH_TIME_FOREVER);
```

Comment:
@ejanzer We had to modify the code in `registerSegmentWithId` and replace it with this one. 2nd argument of `registerBundle` function on `_reactInstance` is no longer a string but the `JSModulesUnbundle`, so in `registerSegmentWithId` we now use `RCTJavaScriptLoader` to load ther bundle either from packager server or filesystem and then `executeApplicationScript` which now can be called multiple times. All of that is wrapped in a `dispatch_group` to make it synchronous to keep compatibility with the old behaviour. The same code was applied in the 3rd PR as well: #27875